sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -3651,6 +3656,27 @@ public ItemDeserializer getItemDeserializer() {
         return this.itemDeserializer;
     }
 
+    @Override
+    public void enableThroughputControl(Set<ThroughputControlGroup> groups) {
+        checkNotNull(groups, "Throughput control groups can not be null");
+
+        if (this.throughputControlEnabled.compareAndSet(false, true)) {
+            ThroughputControlStore throughputControlStore =
+                new ThroughputControlStore(
```

Comment:
Alternatively we could pass "this" to the ThroughputControlStore and remove the dependency on CosmosAsyncContainer from the "groups".

Reply:
We will need the container info so we can query the throughput for the container, since we allow the customer to use throughputThreshold.